### DISCUSSION TITLE
SelectionList empty

### DISCUSSION THREAD

**Discussion by AndyDralle** [Textualize/textual#5415] [Q&A] [answered]
Hey!

After a look at the tutorial, I'm trying to create a minimal program with a `SelectionList`. This is my current code: 

```Python
from textual.app import App, ComposeResult
from textual.widgets import Footer, Header, SelectionList
from textual.widgets.selection_list import Selection

class Tags(SelectionList[int]):
    def compose(self) -> ComposeResult:
        yield SelectionList[int](
            # *[(f"Option: {i}", i) for i in range(10)]
            Selection("Wat", 0, True),
            Selection("Meh", 1)
        )

class ExampleApp(App):
    """A tui with a SelectionList."""

    def compose(self) -> ComposeResult:
        yield Header()
        yield Tags()
        yield Footer()

if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

This gives me the following output: 
![image](https://github.com/user-attachments/assets/9eeb7ff0-83fc-46b1-8427-e6c5b038a18f)

As you can see, my options don't appear. 

If I run the `SelectionList` example from https://textual.textualize.io/widgets/selection_list/#__tabbed_1_2, things work as intended. But I don't want my entire app to be a `SelectionList` like in that example. I just want this to be one of multiple widgets in my tui, like in the tutorial. That's why I'm trying to adapt the tutorial code, instead of just copying the `SelectionList` example directly.

Can somebody spot what I'm missing here? 
Thanks in advance!

**Comment by TomJGooding** [comment, marked as answer]
The problem is that you're trying to compose a `SelectionList` inside another `SelectionList`, which isn't going to work.

If you want your widget to be a subclass of `SelectionList`, you can initialize it using `super().__init__()` and pass in your options:

```python
class MySelectionList(SelectionList):
    def __init__(self) -> None:
        selections = [("First", 1), ("Second", 2)]
        super().__init__(*selections)
```

**Reply by AndyDralle** [reply]
That did the trick. Thanks :)
